controller/src/main/java/org/jboss/as/controller/SimpleResourceDefinition.java

```diff
@@ -462,4 +461,136 @@ protected boolean isOrderedChildResource() {
+         * {@link SimpleResourceDefinition#registerOperations(ManagementResourceRegistration)}
+         *
+         * @param removeHandler the add handler to use.
+         * @return this Parameters object
+         */
+        public Parameters setRemoveHandler(OperationStepHandler removeHandler) {
+            this.removeHandler = removeHandler;
+            return this;
+        }
+
+        /**
+         * Sets the add restart level. The default is {@link OperationEntry.Flag#RESTART_NONE}
+         *
+         * @param addRestartLevel the restart level
+         * @return this Parameters object
+         * @throws IllegalStateException if a null {@code addRestartLevel} is used
+         */
+        public Parameters setAddRestartLevel(OperationEntry.Flag addRestartLevel) {
+            if (addRestartLevel == null) {
+                throw ControllerLogger.ROOT_LOGGER.nullVar("addRestartLevel");
+            }
+            this.addRestartLevel = addRestartLevel;
+            return this;
+        }
+
+        /**
+         * Sets the remove restart level. The default is {@link OperationEntry.Flag#RESTART_ALL_SERVICES}
+         * @param removeRestartLevel the restart level
+         * @return this Parameters object
+         * @throws IllegalStateException if a null {@code addRestartLevel} is used
+         */
+        public Parameters setRemoveRestartLevel(OperationEntry.Flag removeRestartLevel) {
+            if (removeRestartLevel == null) {
+                throw ControllerLogger.ROOT_LOGGER.nullVar("addRestartLevel");
+            }
+            this.removeRestartLevel = removeRestartLevel;
+            return this;
+        }
+
+        /**
+         * Call to indicate that a resource is runtime-only. If not called, the default is {@code false}
+         *
+         * @return this Parameters object
+         */
+        public Parameters setRuntime() {
+            this.runtime = true;
+            return this;
+        }
+
+        /**
+         * Call to deprecate the resource
+         *
+         * @param deprecationData Information describing deprecation of this resource.
+         * @return this Parameters object
+         * @throws IllegalStateException if the {@code deprecationData} is null
+         */
+        public Parameters setDeprecationData(DeprecationData deprecationData) {
+            if (deprecationData == null) {
+                throw ControllerLogger.ROOT_LOGGER.nullVar("addRestartLevel");
+            }
+
+            this.deprecationData = deprecationData;
+            return this;
+        }
+
+        /**
+         * Call to indicate that a resource is of a type where ordering matters amongst the siblings of the same type.
+         * If not called, the default is {@code false}.
+         *
+         * @return this Parameters object
+         */
+
+        public Parameters setOrderedChild() {
+            this.orderedChildResource = true;
+            return this;
+        }
```

Comment:
we could probably add build() method here as well, that could help create resource definition directly.
it could be useful when you don't extend the class